doc/src/arch/reference.rst

```diff
@@ -1897,6 +1902,98 @@ The clocking configuration is specified with ``<clock>`` tags within the ``<cloc
+
+.. arch:tag:: <rib metal_layer="string" y="integer" startx="integer" endx="integer" repeatx="integer" repeaty="integer"/>
+    :req_param metal_layer: A referenced metal layer that sets the unit resistance and capacitance of the distribution wire over the length of the wire.
+    :req_param startx: The start x grid location, of the wire which runs parallel to the x-axis from startx and ends at endx, inclusive. Value can be relative to the device size.
+    :req_param endx: The end of x grid location of the wire. Value can be relative to the device size.
+    :req_param y: The location of the rib with respect to the y-axis. Value can be relative to the device size.
+    :opt_param repeatx: The horizontal repeat factor of the rib along the device. Value can be relative to the device size.
+    :opt_param repeaty: The vertical repeat factor of the rib along the device. Value can be relative to the device size.
+
+Along each spine and rib is a group of switch points. Switch points are used to describe drive or tap locations along the clock distribution wire:
+
+.. arch:tag:: <switch_point type="{drive | tap}" name="string" yoffset="integer" xoffset="integer" xinc="integer" yinc="integer" buffer="string">
+    :req_param type:
+        * ``drive`` -- Drive points are where the clock distribution wire can be driven by a routing switch or buffer.
+        * ``tap`` --  Tap points are where it can drive a routing switch or buffer to send a signal to a different ``clock_network`` or logicblock.
+    :opt_param xoffset: (Only for ``rib`` network) Offset from the ``startx`` of a rib network.
+    :opt_param yoffset: (Only for ``spine`` network) Offset from the ``starty`` of a spine network.
+    :opt_param xinc: (Only for rib ``tap`` points) Descibes the repeat factor of a series of evenly spaced tap points.
+    :opt_param yinc: (Only for spine ``tap`` points) Descibes the repeat factor of a series of evenly spaced tap points.
+    :req_param buffer: (Required only for ``drive`` points) A reference to a pre-defined routing switch. This switch will be used at the drive point. The clock generator uses two buffers to drive the two portions of this ``clock_network`` wire when it is split at the drive point.
+
+Lastly the ``<clock_routing>`` element consists of a group of ``tap`` elements that each separately describe the connectivity from one routing resource (pin or wire) to another. The tap element and its attribute sare as follows:
+
+.. arch:tag:: <tap from="string" to="string" locationx="integer" locationy="integer" switch="string" fc_val="float">
+    :req_param from: The routing resource to make a connection from. The referecing schema can be seen in the clock network exambple where the unique clock network name is forllowe by a period and then the switch name. In addition a special literal ``"ROUTING"`` can be used to reference a connection from the general inter-block routing.
```

Comment:
Probably want to expand this a bit more formally illustrating the two cases:
 * `clock_name.tap_name` format, and
 * `ROUTING` special value